aiida/schedulers/datastructures.py

```diff
@@ -578,30 +590,42 @@ def deserialize_field(self, value, field_type):
         if field_type is None:
             return value
 
-        deserializer_method = getattr(self, '_deserialize_{}'.format(field_type))
+        deserializer_method = getattr(cls, '_deserialize_{}'.format(field_type))
 
         return deserializer_method(value)
 
     def serialize(self):
         """
-        Serialise the current data
-        :return: A serialised representation of the current data
+        Serialize the current data (as obtained by ``self.get_dict()``) into a JSON string.
+
+        :return: A string with serialised representation of the current data.
         """
         from aiida.common import json
 
-        ser_data = {k: self.serialize_field(v, self._special_serializers.get(k, None)) for k, v in self.items()}
+        return json.dumps(self.get_dict())
 
-        return json.dumps(ser_data)
+    def get_dict(self):
+        """
+        Serialise the current data into a dictionary that is JSON-serializable.
+
+        :return: A dictionary
+        """
+        return {k: self.serialize_field(v, self._special_serializers.get(k, None)) for k, v in self.items()}
+
+    def load_from_dict(self, data):
```

Comment:
Wouldn't this make more sense as a classmethod?

Reply:
Done